API/Controllers/TagController.cs

```diff
@@ -0,0 +1,220 @@
+        /// </summary>
+        /// <returns>This method return the tag resource result.</returns>
+        /// <response code="200">This endpoint returns the tag with the specified name.</response>
+        /// <response code="400">The 400 Bad Request status code is returned when the specified tag name is invalid.</response>
+        /// <response code="404">The 404 Not Found status code is returned when no tag is found with the specified tag name.</response>
+        [HttpGet("name/{name}")]
+        [ProducesResponseType(typeof(TagOutput), (int) HttpStatusCode.OK)]
+        [ProducesResponseType(typeof(ProblemDetails), (int) HttpStatusCode.BadRequest)]
+        [ProducesResponseType(typeof(ProblemDetails), (int) HttpStatusCode.NotFound)]
+        public async Task<IActionResult> GetTagByName(string name)
+        {
+            Tag tag = await tagService.FindByNameAsync(name)
+                                         .ConfigureAwait(false);
+            if(tag == null)
+            {
+                ProblemDetails problem = new ProblemDetails
+                {
+                    Title = "Failed getting tag.",
+                    Detail = "The tag could not be found in the database.",
+                    Instance = "FD169337-D328-4E5B-977F-7B2E73995D33"
+                };
+                return NotFound(problem);
+            }
+
+            return Ok(mapper.Map<Tag, TagOutput>(tag));
+        }
+
+        /// <summary>
+        ///     This method is responsible for creating the tag.
+        /// </summary>
+        /// <param name="tagResource">The tag resource which is used to create a tag.</param>
+        /// <returns>This method returns the created tag resource result.</returns>
+        /// <response code="201">This endpoint returns the created tag.</response>
+        /// <response code="400">The 400 Bad Request status code is returned when unable to create tag.</response>
+        [HttpPost]
+        [ProducesResponseType(typeof(TagOutput), (int) HttpStatusCode.Created)]
+        [ProducesResponseType(typeof(ProblemDetails), (int) HttpStatusCode.BadRequest)]
+        public async Task<IActionResult> CreateTagAsync([FromBody] TagInput tagResource)
+        {
+            if(tagResource == null)
+            {
+                ProblemDetails problem = new ProblemDetails
+                {
+                    Title = "Failed to create a new tag.",
+                    Detail = "The specified tag resource was null",
+                    Instance = "5ABFDFE4-7151-46AE-BD2C-32804882E796"
+                };
+                return BadRequest(problem);
+            }
+
+            if(tagResource.Name.Length > 30)
+            {
+                ProblemDetails problem = new ProblemDetails
+                {
+                    Title = "Maximum amount of tag name characters exceeded.",
+                    Detail = "It is not possible to create a project name with more than 75 characters.",
```

Comment:
The method checks on 30 characters length, but the error detail message says 75 characters?

Reply:
Woops, changed it!

Reply:
It also says name and not tags.